htdocs/user/class/user.class.php

```diff
@@ -1653,12 +1653,12 @@ function send_password($user, $password='', $changelater=0)
 
 		// Define $urlwithroot
 		//$urlwithouturlroot=preg_replace('/'.preg_quote(DOL_URL_ROOT,'/').'$/i','',trim($dolibarr_main_url_root));
-		$urlwithroot=$urlwithouturlroot.DOL_URL_ROOT;		// This is to use external domain name found into config file
+		//$urlwithroot=$urlwithouturlroot.DOL_URL_ROOT;		// This is to use external domain name found into config file
```

Comment:
If i compare code at other places that need the external url (from internet), I think the fix is just to uncomment the previous line
$urlwithouturlroot=preg_replace('/'.preg_quote(DOL_URL_ROOT,'/').'$/i','',trim($dolibarr_main_url_root))

Did you try this ?